workspaces/cli-server/src/spectacle/index.ts

```diff
@@ -109,6 +112,28 @@ export class LocalCliSpecRepository implements IOpticSpecReadWriteRepository {
       return events;
     }
   }
+
+  async resetToCommit(batchCommitId: string): Promise<void> {
+    const specJsFile = path.join(
+      this.dependencies.specDirPath,
+      'specification.json'
+    );
+    const rawEventStream = OpticEngineNative.readSpec({
+      specDirPath: this.dependencies.specDirPath,
+    });
+
+    const specEvents = Streams.SpecEvents.fromJSONStream()(rawEventStream);
+    const filteredEvents = pipe(
+      Streams.SpecEvents.takeBatchesUntil(batchCommitId),
+      AsyncTools.intoJSONArray
+    )(specEvents);
+
+    const writeableStream = fs.createWriteStream(specJsFile);
+
+    for await (const chunk of filteredEvents) {
+      writeableStream.write(chunk);
```

Comment:
I'm not sure if this is correct, but this is how I managed to pipe the stream into the writeable stream.

I am a little concerned that this function will exit / the promise will resolve before the last write can occur. I also don't know if there's any race conditions since we aren't waiting for each write to complete (but i presume this _should_ work because its a stream)?

Any other ways to do this? Or is this fine as it is?

Reply:
Thanks for the detailed explanation, I updated with your suggestions above